### DISCUSSION TITLE
[Feat] Warn when deleting query used by alert condition

### DISCUSSION THREAD

**Discussion by Dominik-K** [grafana/grafana#36989] [Alerting]
I have a panel with 4 metric queries (A--D) and alert conditions on each of the queries.

- What did you do?
Deleted the last query (D).

- What was the expected result?
Get a notice that this query is part of the alert and showing all the alert conditions. Question: Are you sure. If yes, delete the alert condition as well.

- What happened instead?
No hint shown. The corresponding alert conditions stays there and the alert's test rule fails:

> Validation failed
>     Failed to extract alert.Reason: Alert on PanelId: 2 refers to query(D) that cannot be found

version: Grafana v4.2.0 (commit: 349f3eb) (Docker image)
datasource: Prometheus

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
